// mongo-glob-economy/app.js
const { Database } = require('quickmongo');
const Discord = require('discord.js')
var url

class Economy {

    /** 
     * @param {string} dbUrl - Mongo Database URL 
     */

    static async setURL(dbUrl) {
        if (!dbUrl) throw new TypeError('Mongo Database URL was not provided');

        return url = dbUrl
    }

  /**
   * @param {string} guildID - define the ID of the guild
   * @param {string} verifiedRoleID - ID of the verified role
   * @param {string} channelID - ID of the verify channel
   * @param {Number} setup - mahh
   */


  static async setVerification(guildID, verifiedRoleID, channelID, setup){
      if(!guildID) throw new TypeError("Guild ID not defined")
      if(!verifiedRoleID) throw new TypeError("Role ID not provided")
      if(!channelID) throw new TypeError("Channel ID not provided")
      if(!setup) throw new TypeError("True or false not set or whatever")
      const db = new Database(url);
      await db.set(`${guildID}.guildID`, guildID)
      await db.set(`${guildID}.verifiedRole`, verifiedRoleID)
      await db.set(`${guildID}.channel`, channelID)
      await db.set(`${guildID}.setup`, setup)
 

  }

  /**
   * @param {string} guildID - ID of the guild
   */

  static async getRoleID(guildID){
      if(!guildID) throw new TypeError("No GUILDID provided!")
      const db = new Database(url)
      const guild = await db.fetch(guildID)
      if(!guild) throw new TypeError("That guild is not in the database");
      const id = await db.fetch(`${guildID}.verifiedRole`)
      return id;


  }

  /**
   * @param {string} guildID - ID of the guild
   */

   static async getChannelID(guildID){
    if(!guildID) throw new TypeError("No GUILDID provided!")
    const db = new Database(url)
    const guild = await db.fetch(guildID)
    if(!guild) throw new TypeError("That guild is not in the database");
    const id = await db.fetch(`${guildID}.channel`)
    return id;


}

/**
   * @param {string} guildID - ID of the guild
   */

 static async getSetup(guildID){
    if(!guildID) throw new TypeError("No GUILDID provided!")
    const db = new Database(url)
    const guild = await db.fetch(guildID)
    if(!guild) throw new TypeError("That guild is not in the database");
    const id = await db.fetch(`${guildID}.setup`)
    return id;


}





}

module.exports = Economy
